core/src/main/scala/stainless/ast/Expressions.scala

```diff
@@ -273,6 +273,13 @@ trait Expressions extends inox.ast.Expressions with Types { self: Trees =>
     }
   }
 
+
+  sealed case class Cell(elem: Expr, base: Type) extends Expr with CachingTyped {
+    override protected def computeType(using s: Symbols): Type = {
+      checkParamType(elem.getType, base, unveilUntyped(CellType(base)))
+    }
+  }
+
```

Comment:
I think it's easier to not have a `Cell` expression but rather rely on `Cell` construction, update, etc. be regular `Class`-related expression trees.
Otherwise, the logic for updating a field, alias analysis and so on would need to be duplicated.

Reply:
Sounds good :) I'm removing all these 👍